src/proof_rules/induction.rs

```diff
@@ -81,6 +84,19 @@ impl Encoding for InvariantAnnotation {
         resolve.visit_expr(invariant)
     }
 
+    fn check_calculus(&self, calculus: &Calculus, direction: Direction) -> Result<(), ()> {
+        if direction
```

Comment:
The inline match expression inside an if statement should be rewritten to just a single match statement that either returns an `Err` or `Ok` without the if then else.